src/DataManagement/Saving/SavingCalculator.cs

```diff
@@ -4,113 +4,144 @@
+            Goal = goal;
+            StartingBalance = startingBalance;
+            EnumCount = Enum.GetValues(typeof(Importance)).Length;
+            CalculateMonthlyIncome();
+        }
 
-            var listToUse = generateRandomData ? expensesRandomList : entryList;
-            var groupedByImportance = (from entry in listToUse
-                                       where entry.Importance != 1
-                                       group entry by entry.Importance).ToArray();
+        public CalculationResults GetSuggestedExpensesOffers(SavingType savingType = SavingType.Regular)
+        {
+            SavingType = savingType;
+            EntrySuggestions = new List<ISavingSuggestion>();
+            SavingSuggestionByImportanceList = new List<SavingSuggestionByImportance>();
+            GetAverages();
+            CalculateSavingTime();
+            return new CalculationResults(EntrySuggestions, SavingSuggestionByImportanceList, MonthsToSave);
+        }
 
-            foreach (var group in groupedByImportance)
+        private void CalculateMonthlyIncome()
+        {
+            if (IncomeList.Any())
             {
-                var i = group.Key;
-                var expenses = group.ToIEntryList();
-
-                var ratio = enumCount - i;
-                foreach (var entry in expenses)
+                var newest = IncomeList.GetNewestEntryDate();
+                var oldest = IncomeList.GetOldestEntryDate();
+                var months = 0;
+                while (oldest <= newest)
                 {
-                    decimal amountAfterSaving;
-                    switch (savingType)
-                    {
-                        case SavingType.Minimal:
-                            if (entry.Amount * ratio * MinimalSavingValue < entry.Amount)
-                            {
-                                amountAfterSaving = entry.Amount * ratio * MinimalSavingValue;
-                            }
-                            else
-                            {
-                                amountAfterSaving = entry.Amount;
-                            }
-                            break;
-                        case SavingType.Regular:
-                            amountAfterSaving = entry.Amount * ratio * RegularSavingValue;
-                            break;
-                        case SavingType.Maximal:
-                            amountAfterSaving = entry.Amount * ratio * MaximalSavingValue;
-                            break;
-                        default:
-                            throw new Exception("Unexpected saving type");
-                    }
-                    entrySuggestions.Add(new SavingSuggestion(entry, amountAfterSaving));
-
-                    sumsOfAmountByImportanceAdjusted[i - 1] += amountAfterSaving;
-                    sumsOfAmountByImportanceDefault[i - 1] += entry.Amount;
-                    entryAmounts[i - 1]++;
+                    months++;
```

Comment:
Maybe this could be calculated by just Newest - Oldest (by converting years to months and adding months) ?

Reply:
Yeah this is a silly way to do it :D I'll fix this